Hakawai/Core/HKWTextView.m

```diff
@@ -121,11 +128,52 @@ - (id)awakeAfterUsingCoder:(__unused NSCoder *)aDecoder {
     return replacement;
 }
 
+- (void)textStorage:(__unused NSTextStorage *)textStorage
+  didProcessEditing:(__unused NSTextStorageEditActions)editedMask
+              range:(NSRange)editedRange
+     changeInLength:(NSInteger)delta {
+    if (!enableKoreanMentionsFix) {
+        // If this mentions fix is not enabled, don't do anything in text storage
+        return;
+    }
+
+    if (delta > 0) {
+        // If the delta is greater than 0, this is an insertion
+        NSString *change = [self.text substringFromIndex:self.text.length-(NSUInteger)delta];
+        [self.controlFlowPlugin textView:self
+                 shouldChangeTextInRange:NSMakeRange(editedRange.location, 0)
+                         replacementText:change
+                             isInsertion:true];
+        // Update the saved text state so that it can be accessed in the case of deletion
+        if (self.textStateBeforeDeletion.length == 0) {
+            self.textStateBeforeDeletion = change;
+        } else {
+            self.textStateBeforeDeletion = [self.textStateBeforeDeletion stringByAppendingString:change];
+        }
+    }
+    else if (delta < 0) {
+        // If the delta is less than 0, this is a deletion
+        NSUInteger absoluteDelta = (NSUInteger)labs((long)delta);
+        NSUInteger differential = self.textStateBeforeDeletion.length-absoluteDelta;
```

Comment:
Should we add a sanity check for the `differential` to not overflow by ensuring `textStateBeforeDeletion` is not smaller than `absoluteDelta`?